src/main/kotlin/com/openlattice/mechanic/upgrades/SetOriginIdToNonNullUpgrade.kt

```diff
@@ -0,0 +1,38 @@
+package com.openlattice.mechanic.upgrades
+
+import com.google.common.base.Stopwatch
+import com.openlattice.IdConstants
+import com.openlattice.mechanic.Toolbox
+import com.openlattice.postgres.PostgresColumn.ORIGIN_ID
+import com.openlattice.postgres.PostgresDataTables
+import com.openlattice.postgres.PostgresTable.DATA
+import org.slf4j.LoggerFactory
+import java.util.concurrent.TimeUnit
+
+class SetOriginIdToNonNullUpgrade(private val toolbox: Toolbox) : Upgrade {
+
+    companion object {
+        private val logger = LoggerFactory.getLogger(Toolbox::class.java)
+    }
+
+    override fun getSupportedVersion(): Long {
+        return Version.V2019_08_26.value
+    }
+
+    override fun upgrade(): Boolean {
+
+        val updateQuery = "UPDATE ${DATA.name} SET ${ORIGIN_ID.name} = ${IdConstants.EMPTY_ORIGIN_ID.id}::uuid WHERE ${ORIGIN_ID.name} = NULL"
+        toolbox.rateLimitedQuery( 16, updateQuery, logger)
+
+        val pkey = PostgresDataTables.buildDataTableDefinition().primaryKey.joinToString(",") { it.name }
+        val dropPkey = "ALTER TABLE ${DATA.name} DROP CONSTRAINT ${DATA.name}_pkey"
+        val updatePkey = "ALTER TABLE ${DATA.name} ADD PRIMARY KEY ($pkey)"
```

Comment:
would probably also be good to set the default value for ORIGIN_ID

Reply:
It's in a separate PR but since we have to deploy this one pre-migration, sense is made

Reply:
I feel like we should run this before the main migration though, since it'll probably take a while to run and new entities will likely be created while it's running, so if we don't set a default value beforehand and try to update the primary key after there's a good chance it'll fail since there will probably be new null ORIGIN_ID values

Reply:
Right, sorry. What I meant to communicate is that the PR that will be deployed pre-this-migration now includes setting the default value 👍

Reply:
OOOOOOOHHHHHHHHH lol I see